src/controllers/authController.js

```diff
@@ -0,0 +1,48 @@
+import passport from 'passport';
+import { generateUserToken } from '../utils';
+import passportConfIgnored from '../config/passport';
+
+/**
+   * It receives the social provider to use for authentication and returns an express
+   * controller function to handle the callback request that happens after an
+   * authentication attempt has happened
+   * @function callback
+   * @param {string} provider
+   * @returns {function} callbackFunction
+*/
+export const callback = provider => (req, res, next) => {
+  passport.authenticate(
+    provider,
+    (err, user) => {
+      // check if server returns an error
+      if (err) return res.status(500).json({ status: 500, message: 'Registration unsuccessful. Please try again later.' });
+      // check if user authentication failed on social platform
+      // eslint-disable-next-line no-else-return
```

Comment:
@eokwukwe I had to use inline rule here because `eslint` thinks I should not write an `else` statement after a `return` statement (which I ordinarily won't do too), but if I take `eslint`'s advice and use two different `if` statements, it increases the `cognitive complexity`, and `codeclimate` complains. Yet I have to check for both conditions and return different `status codes` and `messages` to the user, and I don't want to set an application-wide rule in the `.eslintrc` file

Reply:
You don't need the `else if` block because the callback will return either `error` or `user`. The check for `err` is the same as the check for `!user`. Hence all you need to check is for error. The error will be returned if the user uses an invalid credential for the third party authentication. Your code should look like this:

```
if (err) return res.status(500).json({ status: 500, message: 'Registration unsuccessful. Please try again later.' });

// process successfully authenticated and registered user
const token = generateUserToken(user);
res.status(res.statusCode).json({ status: res.statusCode, token });
```

or if you still want to check for the user, you may write it this way:
```
if (err || !user) return res.status(500).json({ status: 500, message: 'Registration unsuccessful. Please try again later.' });

// process successfully authenticated and registered user
const token = generateUserToken(user);
res.status(res.statusCode).json({ status: res.statusCode, token });
```

Reply:
Can I use the second? Because in my test, to increase the coverage, I am actually testing for both scenarios differently. The error test is detecting error from the database server, while the null user is detecting error from authentication from the social media platform..

Reply:
Yes you can

Reply:
Done.